sunpy/util/scraper.py

```diff
@@ -231,8 +232,11 @@ def filelist(self, timerange):
                     soup = BeautifulSoup(opn, "html.parser")
                     for link in soup.find_all("a"):
                         href = link.get("href")
-                        if href.endswith(self.pattern.split('.')[-1]):
-                            fullpath = directory + href
+                        if href is not None and href.endswith(self.pattern.split('.')[-1]):
+                            if href[0] == '/':
+                                fullpath = self.domain + href[1:]
+                            else:
+                                fullpath = directory + href
                             if self._URL_followsPattern(fullpath):
```

Comment:
Are these changes are bug fix or just specific to this client?

Reply:
It is a bugfix. Luckily in all our previous clients, we don't check whether `href` can be `None`; because those html pages opened by `BeautifulSoup` never had one.

Another fix is about relative paths of file returned by href, the directory opened by scraper was like `https://some_domain/pub/keep/2016/05/18` and href returned something like `/pub/keep/2016/05/18/file_nameHHMMSS.fts` so full url becomes `https://some_domain/pub/keep/2016/05/18/pub/keep/2016/05/18/file_nameHHMMSS.fts` which is wrong, and it should be just `domain + href[1:]`. Checking whether file href starts from `\` to know href are relative to just domain, not the directory.

Reply:
Could this be spun off into its own PR?

Reply:
Ok